light-client/src/components/verifier.rs

```diff
@@ -26,35 +24,29 @@ impl From<Result<(), VerificationError>> for Verdict {
     }
 }
 
+/// The verifier checks:
+///
+/// a) whether a given untrusted light block is valid, and
+/// b) whether a given untrusted light block should be trusted
+///    based on a previously verified block.
+///
+/// ## Implements
+/// - [TMBC-VAL-CONTAINS-CORR.1]
+/// - [TMBC-VAL-COMMIT.1]
 pub trait Verifier {
-    fn verify(
-        &self,
-        light_block: &LightBlock,
-        trusted_state: &TrustedState,
-        options: &Options,
-    ) -> Verdict {
-        self.validate_light_block(light_block, trusted_state, options)
-            .and_then(|| self.verify_overlap(light_block, trusted_state, options))
-            .and_then(|| self.has_sufficient_voting_power(light_block, options))
-    }
-
-    fn validate_light_block(
-        &self,
-        light_block: &LightBlock,
-        trusted_state: &TrustedState,
-        options: &Options,
-    ) -> Verdict;
-
-    fn verify_overlap(
-        &self,
-        light_block: &LightBlock,
-        trusted_state: &TrustedState,
-        options: &Options,
-    ) -> Verdict;
-
-    fn has_sufficient_voting_power(&self, light_block: &LightBlock, options: &Options) -> Verdict;
+    /// Perform the verification.
+    fn verify(&self, untrusted: &LightBlock, trusted: &LightBlock, options: &Options) -> Verdict;
 }
 
+/// Production implementation of the verifier.
+///
+/// For testing purposes, this implementation is parametrized by:
+/// - A set of predicates used to validate a light block
+/// - A voting power calculator
+/// - A commit validator
+/// - A header hasher
+///
+/// For regular use, one can summon an implementation with `ProdVerifier::default()`.
```

Comment:
s/summon/construct? 🧙

Reply:
Right, the default instance is indeed not fancy enough to warrant such witchcraft ;)